File: binary_search/binary_search.py
import unittest

def binary_search(i, l):
    """
    Checks if an item exist in a list using a non-recursive implementation of binary search

    i -- item to search for
    l -- item to search in 
    """
    first = 0
    last = len(l) - 1
    found = False

    while first <= last and found is False:
        midpoint = (first + last) // 2
        if l[midpoint] == i:
            return True
        else:
            if i < l[midpoint]:
                last = midpoint-1
            else:
                first = midpoint+1


# Tests
class TestBinarySearch(unittest.TestCase):
    def test_odd(self):
        self.assertEqual(
            binary_search(9, [1, 3, 5, 7, 9]),
            True
        )

    def test_even(self):
        self.assertEqual(
            binary_search(4, [2, 4, 6, 8, 10, 12]),
            True
        )

if __name__ == '__main__':
    unittest.main()

File: binary_search/binary_search.js
/**
 * Non-recursive version of binary serach 
 * 
 * @param {integer} item item to search for
 * @param {integer[]} list list to search in
 * @returns {boolean} is found or not
 */
function binarySearch(item, list) {
    first = 0;
    last = list.length - 1;
    found = false;

    while (first <= last && found === false) {
        midpoint = parseInt((first + last) / 2, 10);
        if (list[midpoint] == item) {
            return true;
        } else {
            if (item < list[midpoint]) {
                last = midpoint-1;
            } else {
                first = midpoint+1;
            }
        }
    }
}

console.assert(binarySearch(9, [1, 3, 5, 7, 9]), true);
console.assert(binarySearch(4, [2, 4, 6, 8, 10, 12]), true);